lib/iris/tests/test_aggregate_by.py

```diff
@@ -1328,5 +1349,153 @@ def test_weights_fail_with_non_weighted_aggregator(self):
         )
 
 
+# Simply redo the tests of TestAggregateBy with other cubes as weights
+# Note: other weights types are not tested this way here since this would
+# require adding dimensional metadata objects to the cubes, which would change
+# the the CMLs of all resulting cubes
```

Comment:
Actually these tests do check the CML since the test class inherits from `TestAggregateBy`. For other types of weights, this would result in different outputs, and one would need to add new CMLs for each of the tests. Since the usage of other weight types is covered in other tests, I did not do this.

Reply:
Sorry, you're right! I hadn't correctly read the inheritance. So what you've done is correct!